File: project/sungwoong-onboarding/src/main/java/com/multi/sungwoongonboarding/responses/presentation/ResponseController.java
package com.multi.sungwoongonboarding.responses.presentation;

import com.multi.sungwoongonboarding.common.dto.ResponseDto;
import com.multi.sungwoongonboarding.common.util.ResponseUtil;
import com.multi.sungwoongonboarding.responses.application.ResponseService;
import com.multi.sungwoongonboarding.responses.dto.ResponseCreateRequest;
import com.multi.sungwoongonboarding.responses.dto.ResponseSheet;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/responses")
public class ResponseController {

    private final ResponseService responseService;

    @PostMapping
    public ResponseDto<ResponseSheet> submitResponse(@RequestBody @Valid ResponseCreateRequest responseCreateRequest) {
        ResponseSheet responseSheet = responseService.submitResponse(responseCreateRequest);
        return ResponseUtil.success(responseSheet);
    }
}

File: project/soojinhan-onboarding/src/main/kotlin/onboardingproject/project/controller/ResponseController.kt
package onboardingproject.project.controller

import jakarta.servlet.http.HttpServletRequest
import onboardingproject.project.common.dto.ResponseDto
import onboardingproject.project.dto.SaveFieldResponseRequest
import onboardingproject.project.dto.SurveyResponse
import onboardingproject.project.service.ResponseService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

/**
 * packageName : onboardingproject.project.controller
 * fileName    : ResponseController
 * author      : hsj
 * date        : 2025. 6. 21.
 * description :
 */
@RestController
class ResponseController(
    private val responseService: ResponseService
) {

    /**
     * 응답 생성 API
     */
    @PostMapping("/surveys/{surveyId}/responses")
    fun createResponse(
        request: HttpServletRequest,
        @PathVariable surveyId: String,
        @RequestBody saveResponseRequest: List<SaveFieldResponseRequest>,
    ): ResponseDto<Unit> {
        return ResponseDto(
            path = request.servletPath,
            data = responseService.createResponse(saveResponseRequest)
        )
    }


    /**
     * 응답 조회 API
     */
    @GetMapping("/surveys/{surveyId}/responses")
    fun getResponse(
        request: HttpServletRequest,
        @PathVariable surveyId: String
    ): ResponseDto<List<SurveyResponse>> {
        return ResponseDto(
            path = request.servletPath,
            data = responseService.getSurveyResponse(surveyId)
        )
    }
}